src/pixel/state.js

```diff
@@ -52,60 +52,111 @@ import { toParams } from '../utils/url'
 
 const noOpEvents = ['setemail', 'setemailhash', 'sethashedemail']
 
-const _pMap = {
-  appId: aid => {
-    return asStringParam('aid', aid)
-  },
-  eventSource: source => {
-    return asParamOrEmpty('se', source, (s) => base64UrlEncode(JSON.stringify(s, replacer)))
-  },
-  liveConnectId: fpc => {
-    return asStringParam('duid', fpc)
-  },
-  trackerName: tn => {
-    return asStringParam('tna', tn || 'unknown')
-  },
-  pageUrl: purl => {
-    return asStringParam('pu', purl)
-  },
-  errorDetails: ed => {
-    return asParamOrEmpty('ae', ed, (s) => base64UrlEncode(JSON.stringify(s)))
-  },
-  retrievedIdentifiers: identifiers => {
-    const identifierParams = []
-    identifiers.forEach((i) => identifierParams.push(asStringParam(`ext_${i.name}`, i.value)))
-    return identifierParams
-  },
-  hashesFromIdentifiers: hashes => {
-    const hashParams = []
-    hashes.forEach((h) => hashParams.push(asStringParam('scre', `${h.md5},${h.sha1},${h.sha256}`)))
-    return hashParams
-  },
-  decisionIds: dids => {
-    return asStringParam('li_did', dids.join(','))
-  },
-  hashedEmail: he => {
-    return asStringParam('e', he.join(','))
-  },
-  usPrivacyString: usps => {
-    return asStringParam('us_privacy', usps)
-  },
-  wrapperName: wrapper => {
-    return asStringParam('wpn', wrapper)
-  },
-  gdprApplies: gdprApplies => {
-    return asParamOrEmpty('gdpr', gdprApplies, (s) => encodeURIComponent(s ? 1 : 0))
-  },
-  gdprConsent: gdprConsentString => {
-    return asStringParam('gdpr_consent', gdprConsentString)
-  },
-  referrer: referrer => {
-    return asStringParam('refr', referrer)
-  },
-  contextElements: contextElements => {
-    return asStringParam('c', contextElements)
-  }
-}
+const _pArray = [
+  [
+    'appId',
+    aid => {
+      return asStringParam('aid', aid)
+    }
+  ],
+  [
+    'eventSource',
+    source => {
+      return asParamOrEmpty('se', source, (s) => base64UrlEncode(JSON.stringify(s, replacer)))
+    }
+  ],
+  [
+    'liveConnectId',
+    fpc => {
+      return asStringParam('duid', fpc)
+    }
+  ],
+  [
+    'trackerName',
+    tn => {
+      return asStringParam('tna', tn)
```

Comment:
We previously had  `asStringParam('tna', tn || 'unknown')` here. From the code it seems, we never returned `unknown` though. Also the tests started failing when `unknown` was returned as default in case the real value is not present. LP defaults to `lp` when `tna` is empty.